ambari-server/src/main/java/org/apache/ambari/server/state/stack/UpgradePack.java

```diff
@@ -630,113 +552,51 @@ private void initializeTasks(String service, List<Task> tasks) {
-    /**
-     * List of additional prerequisite checks to run in addition to required prerequisite checks
-     */
-    @XmlElement(name="check", type=String.class)
-    public List<String> checks = new ArrayList<>();
-
-    /**
-     * Prerequisite checks configuration
-     */
-    @XmlElement(name="configuration")
-    public PrerequisiteCheckConfig configuration;
+  public PrerequisiteCheckConfig getPrerequisiteCheckConfig() {
+    return new PrerequisiteCheckConfig(prerequisiteChecks);
   }
 
   /**
-   * Prerequisite checks configuration
+   * Holds the config properties for all the checks defined for an Upgrade Pack.
    */
   public static class PrerequisiteCheckConfig {
-    /**
-     * Global config properties common to all prereq checks
-     */
-    @XmlElement(name="property")
-    public List<PrerequisiteProperty> globalProperties;
-
-    /**
-     * Config properties for individual prerequisite checks
-     */
-    @XmlElement(name="check-properties")
-    public List<PrerequisiteCheckProperties> prerequisiteCheckProperties;
-
-    /**
-     * Get global config properties as a map
-     * @return Map of global config properties
-     */
-    public Map<String, String> getGlobalProperties() {
-      if(globalProperties == null) {
-        return null;
-      }
-      Map<String, String> result = new HashMap<>();
-      for (PrerequisiteProperty property : globalProperties) {
-        result.put(property.name, property.value);
-      }
-      return result;
+    private List<PrerequisiteCheckDefinition> m_checks;
+    private PrerequisiteCheckConfig(List<PrerequisiteCheckDefinition> checks) {
+      m_checks = checks;
     }
 
-    /**
-     * Get config properties for a given prerequisite check as a map
-     * @param checkName The prerequisite check name
-     * @return Map of config properties for the prerequisite check
-     */
     public Map<String, String> getCheckProperties(String checkName) {
-      if(prerequisiteCheckProperties == null) {
-        return null;
+      if (null == m_checks) {
+        return Collections.emptyMap();
       }
-      for(PrerequisiteCheckProperties checkProperties : prerequisiteCheckProperties) {
-        if(checkProperties.name.equalsIgnoreCase(checkName)) {
-          return checkProperties.getProperties();
-        }
+
+      Optional<PrerequisiteCheckDefinition> optional = m_checks.stream()
+          .filter(c -> c.className.equals(checkName)).findFirst();
+
+      if (!optional.isPresent()) {
+        return Collections.emptyMap();
       }
-      return null;
+
+      PrerequisiteCheckDefinition checks = optional.get();
+
+      Map<String, String> map = checks.properties.stream().collect(Collectors.toMap(
+          c -> c.name, c -> c.value));
+
+      return map;
     }
   }
 
   /**
-   * Config properties for a specific prerequisite check.
+   * The definition for each check in the Upgrade Pack.
    */
-  public static class PrerequisiteCheckProperties {
-    /**
```

Comment:
Could properties still be useful anymore? If someone writes a generic check class and just wants to change what's passed in for different upgrade packs?

Reply:
They would have to set properties for each class they wanted to customize.  There could be a generic class in Ambari that every UP could use, which always gets run.  But if a UP wanted to customize the properties, they'd have to put it in the XML as a `<check className=...` element.

Reply:
My thought was that it might be useful for mpack owners to write their own generic classes that could be used for more than 1 module.